app/classes/Transvision/ShowResults.php

```diff
@@ -403,4 +403,35 @@ public static function resultsTable($search_results, $recherche, $locale1, $loca
 
         return $table;
     }
+
+    /**
+     * Search entity names: search full entity IDs (including path and filename),
+     * then search entity names (without the full path) if there are no results.
+     *
+     * @param array  $source_strings Array of source strings
+     * @param string $regex          Regular expression to search entity names
+     *
+     * @return array List of matching entity names
+     */
+    public static function searchEntities($source_strings, $regex)
+    {
+        // Search through the full entity ID
+        $entities = preg_grep($regex, array_keys($source_strings));
+
+        /* If there are no results, search also through the entity names.
+         * This is needed for "perfect match" when only the entity name is
+         * provided.
+         */
+        if (! $entities) {
```

Comment:
although it works because php is a dynamically typed language and an empty array is evaluated as false, I think I would prefer that we explicitely check for emptyness with empty()